src/driver/DriverUtils.ts

```diff
@@ -14,27 +14,18 @@ export class DriverUtils {
     static buildDriverOptions(options: any, buildOptions?: { useSid: boolean }): any {
         if (options.url) {
             const parsedUrl = this.parseConnectionUrl(options.url);
+            let urlDriverOptions: any = {
+                type: options.type || parsedUrl.type,
+                host: options.host || parsedUrl.host,
+                username: options.username || parsedUrl.username,
+                password: options.password || parsedUrl.password,
+                port: options.port || parsedUrl.port,
+                database: options.database || parsedUrl.database
+            };
             if (buildOptions && buildOptions.useSid) {
-                const urlDriverOptions: any = {
-                    type: options.type || parsedUrl.type,
-                    host: options.host || parsedUrl.host,
-                    username: options.username || parsedUrl.username,
-                    password: options.password || parsedUrl.password,
-                    port: options.port || parsedUrl.port,
-                    sid: options.sid || options.database || parsedUrl.database
-                };
-                return Object.assign(urlDriverOptions, options);
-            } else {
-                const urlDriverOptions: any = {
-                    type: options.type || parsedUrl.type,
-                    host: options.host || parsedUrl.host,
-                    username: options.username || parsedUrl.username,
-                    password: options.password || parsedUrl.password,
-                    port: options.port || parsedUrl.port,
-                    database: options.database || parsedUrl.database
-                };
-                return Object.assign(urlDriverOptions, options);
-            }
+                urlDriverOptions.sid = options.sid || options.database || parsedUrl.database
+            };
+            return Object.assign(urlDriverOptions, options);
```

Comment:
Maybe
```suggestion
            return Object.assign({}, options, urlDriverOptions);
```
?
Otherwise empty values from options will still override parsed url params.

Reply:
That works even better, now I was able to even simplify the urlDriverOptions further. Please check the revised version.

Reply:
lgtm